agent/src/capability/filteredconfigstore.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "filteredconfigstore.h"
+
+FilteredConfigStore::FilteredConfigStore(const std::string &path): BaseConfigStore(path)
+{
+}
+
+FilteredConfigStore::~FilteredConfigStore()
+{
+}
+
+GatewayConfiguration FilteredConfigStore::configsByID(const std::vector<std::string> &capIDs) const
+{
+    GatewayConfiguration conf;
+    for (auto &id : capIDs) {
+        conf.append(configByID(id));
```

Comment:
Is it not better to return all found things instead of empty when something fails?